sample_annotator/clients/submission_portal_client.py

```diff
@@ -0,0 +1,185 @@
+        quantity_params = ['slope_aspect', 'al_sat', 'annual_precpt', 'annual_temp', 'season_precpt', 'season_temp', 'slope_gradient', 'soil_text_measure', 'temp', 'microbial_biomass', 'carb_nitro_ratio', 'org_matter', 'org_nitro', 'tot_carb', 'tot_nitro_content', 'tot_org_carb', 'tot_phosp', 'phosphate', 'salinity', 'samp_store_temp', 'size_frac_low', 'depth', 'size_frac_up', 'samp_size', 'alt']
+        controlled_term_params = ['env_broad_scale', 'env_local_scale', 'env_medium', 'experimental_factor', 'growth_facil', 'samp_mat_process', 'chem_administration']
+        float_params = ['elev', 'ph']
+        time_params = ['collection_date']
+        geoloc_params = ['lat_lon']
+
+
+        sampledata = response['metadata_submission']['sampleData']
+        db_lis = []
+        for super_keys in sampledata.keys():
+            current_lis = sampledata[super_keys]
+            db = nmdc.Database()
+            id_counter = 0
+            nmdc_ids = self.mint_nmdc_ids(len(current_lis))
+
+            for samp_dic in current_lis:
+                param_dic = {}
+                for key in samp_dic.keys():
+
+                    if(samp_dic[key] == 'nan' or samp_dic[key] == 'null' or samp_dic[key] == None):
+                        continue
+
+                    if (key in string_params):
+                        if (key == 'id'):
+                            param_dic['alternative_identifiers'] = samp_dic[key] #this vlue might need to be different than 'alternative_identifiers'
+                            param_dic[key] = nmdc_ids[id_counter]
+                            id_counter = id_counter + 1
+                            
+                        else:
+                            if(";" in samp_dic[key]):
+                                temp = samp_dic[key].split(";")
+                            else:
+                                temp = samp_dic[key]
+
+                            param_dic[key] = temp
+
+                    elif (key in controlled_term_params):
+
+                        #convert then add to dictionary
+                        if (key == "env_broad_scale" or key == "env_local_scale" or key == "env_medium"):
+                            env_code = samp_dic[key]
+                            env_code = env_code.split("[")[1]
+                            env_code = env_code.replace("]","")
+                            temp = nmdc.ControlledTermValue(term=nmdc.OntologyClass(id=env_code))
+                            param_dic[key] = temp
```

Comment:
This needs to be a [ControlledIdentifiedTermValue](https://microbiomedata.github.io/nmdc-schema/ControlledIdentifiedTermValue/) and not a [ControlledTermValue](https://microbiomedata.github.io/nmdc-schema/ControlledTermValue/). You can what dataclasses to use for slots by looking at this page: https://microbiomedata.github.io/nmdc-schema/Biosample/

Reply:
Ive made this update and the output now looks correct, including the id, name, and raw value.